### PR TITLE
Declination: added script to generate tables

### PR DESCRIPTION
**Pull request by tridge**
this is separated out from #7124

### CODE REVIEW

File: libraries/AP_Declination/generate/generate.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    declination = r2d * atan2(y, x)
+    inclination = r2d * asin(z / intensity)
+    return [intensity, inclination, declination]
+
+isv = 0
+date = datetime.datetime.now()
+itype = 1
+alt = 0.0
+
+lat = -35.362917
```
Comment (WickedShell) [inline libraries/AP_Declination/generate/generate.py]: Is this needed? Looks like it gets set appropriately on the fly later.

**Comment by WickedShell** [comment]
I'm excited to see scripts for this coming back! Has the script been tested against python3 as well?

File: libraries/AP_Declination/generate/generate.py (hunk 1 above)
Comment (tridge) [reply libraries/AP_Declination/generate/generate.py]: fixed

**Comment by tridge** [comment]
no, unfortunately the pyigrf12 package doesn't work on py3 yet

**Comment by WickedShell** [comment]
Really?

```
wickedshell@milton ~ % pip2 install pyigrf12
Collecting pyigrf12
  Downloading pyigrf12-1.2.1.tar.gz
pyigrf12 requires Python '>=3.6' but the running Python is 2.7.14
```

When trying to install for python3 I have to provide a fortran compiler first... Will try and find one and see if I can get that going.

**Comment by WickedShell** [comment]
So I can install pyigrf12 under python3, but not python2, however when I run it I get the following error:
```
wickedshell@milton ..ode/ardupilot/libraries/AP_Declination (git)-[remotes/tridge/pr-mag-tables] % generate/generate.py                             :(
Traceback (most recent call last):
  File "generate/generate.py", line 39, in <module>
    inclination_table = np.empty((NUM_LAT, NUM_LON))
TypeError: 'float' object cannot be interpreted as an integer
```

So I guess the generation script is not to be used by me :(

**Pull request merged**